sdk/core/System.ClientModel/src/Message/PipelineMessage.cs

```diff
@@ -32,6 +32,23 @@ protected internal PipelineMessage(PipelineRequest request)
         protected internal set;
     }
 
+    public PipelineResponse ExtractResponse()
+    {
+        AssertResponse();
```

Comment:
why do we need to assert that? Why not simply return null when there is no response. As implemented now, the caller has to always expect an exception.

Reply:
Fair.  I had also gone back and forth on this and didn't know if we wanted to make the return type nullable.  But I have been studying the response.Content thread and given in .NET, we provide testers to avoid throwing an exception, I agree now that it would be better not to throw in this case and just return null if there is no response.  It is a low-level API so it should be fine to require the caller to do a null-check on the return value.